packages/richtext-lexical/src/features/link/server/baseFields.ts

```diff
@@ -117,13 +117,20 @@ export const getBaseFields = (
       type: 'relationship',
       filterOptions:
         !enabledCollections && !disabledCollections
-          ? ({ relationTo, user }) => {
-              const hidden = config.collections.find(({ slug }) => slug === relationTo)?.admin
-                .hidden
+          ? async ({ relationTo, req, user }) => {
+              const admin = config.collections.find(({ slug }) => slug === relationTo)?.admin
+              const baseFilter = await (admin?.baseFilter ?? admin?.baseListFilter)?.({
+                limit: 0,
+                page: 1,
+                req,
+                sort: 'id',
+              })
+
+              const hidden = admin?.hidden
               if (typeof hidden === 'function' && hidden({ user } as { user: TypedUser })) {
                 return false
               }
-              return true
+              return baseFilter ?? true
```

Comment:
This is the key part were the bug in Lexical it is solved

Reply:
Nice! 

Not sure I love this part
```ts
await (admin?.baseFilter ?? admin?.baseListFilter)?.
```
What about this:

```ts
const admin = config.collections.find(({ slug }) => slug === relationTo)?.admin

const hidden = admin?.hidden
if (typeof hidden === 'function' && hidden({ user } as { user: TypedUser })) {
  return false
}

let baseFilter = admin?.baseFilter  ?? admin?.baseListFilter
if (baseFilter) {
  return baseFilter({
    limit: 0,
    page: 1,
    req,
    sort: 'id',
  })
}

return true
```

This also allows the function to not run if the field is hidden.